packages/sanity-plugin-nrkno-spellcheck/README.md

```diff
@@ -0,0 +1,354 @@
+
+_Figure 5: Toast showing number of corrections applied._
+
+Because the document can change while a users is editing corrections, a best-effort approach to edits is used. Only patches that can be safely applied to a field, array-item or portable-text block will be committed. 
+
+Eg: If a title field and two blocks in a portable text field are corrected, and one of the text blocks was concurrently edited, only the title and second text block would be patched by nrkno-spellchecker.
+
+![partial-correction.png](docs/images/partial-correction.png)
+
+_Figure 6: Toast showing some discarded corrections._
+
+### Error handling
+
+When and if the configured spellcheck service throws an error during processing,
+the user will be informed. If the user choose to retry, only failed batches will be reprocessed.
+
+![error-handling.png](docs/images/error-handling.png)
+
+_Figure 7: Popover showing that some spellcheck service batches failed._
+
+
+# Installation
+
+In Sanity studio project run:
+
+`npx sanity install @nrk/sanity-plugin-nrkno-spellcheck`
+
+This will run yarn install & add the plugin to sanity.json plugin array.
+
+# Usage
+
+It is recommended to use this library with 
+[@nrk/sanity-plugin-nrkno-odd-utils](https://github.com/nrkno/nrkno-sanity-libs/blob/master/packages/sanity-plugin-nrkno-odd-utils),
+so the built-in sanity document component can be easily decorated with a spellcheck button.
+
+## Basic setup
+
+Creates and configures ConfiguredSpellcheckButton with default the default word-parser and a dummy spellcheck-service.
+
+Given the following dummy-spellcheck-service:
+
+```ts
+//dummy-spellcheck-service.ts
+import {
+  Language,
+  SpellcheckResponse,
+  SpellcheckService,
+} from '@nrk/sanity-plugin-nrkno-spellcheck';
+
+const knownWords = ['slim', 'pickings'];
+export const spellcheckService: SpellcheckService = async function dummy(
+  words: string[],
+  language: Language
+): Promise<SpellcheckResponse[]> {
+  return words
+    .filter((word) => !suggestions.includes(word))
+    .map((word) => ({ word, suggestions}));
+};
+
+``` 
+
+Create the following React component:
+```tsx
+import React, { forwardRef, Ref } from 'react';
+import { ObjectSchemaType, SanityDocument } from '@sanity/types';
+import { NestedFormBuilder, unsetInputComponent } from '@nrk/sanity-plugin-nrkno-odd-utils';
+import {
+  CachedSpellchecker,
+  ConfiguredSpellcheckButton,
+  createWordParser,
+  defaultDisplayTexts,
+  DisplayTexts,
+  DisplayTextsContext,
+} from '@nrk/sanity-plugin-nrkno-spellcheck';
+import { languages, spellcheckService } from './dummy-spellcheck-service';
+
+// running spellcheck twice for the same language will use cached suggestions
+// this instance can be reused between components
+const spellchecker = new CachedSpellchecker({ spellcheckService });
+
+// default: words less than 3 letters, in all caps or with numbers are skipped
+const wordParser = createWordParser(/*{ config is optional }*/);
+
+const dummyLanguage = {
```

Comment:
Skulle denne kanskje ha brukt `languages` fra dummy-spellcheck-service? Evt at man bare passer languages direkte inn til `language`-propen for `ConfiguredSpellcheckButton`?